File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/digitaltwin/InvokeOperationRequest.java
package org.opcfoundation.webserver.service.message.digitaltwin;

import org.eclipse.milo.opcua.stack.core.types.builtin.Variant;
import org.opcfoundation.webserver.types.digitaltwin.ObjectServiceContext;

import java.util.List;

public class InvokeOperationRequest {
    private final String id;
    private final String operationName;
    private final List<Variant> inputArguments;
    private final ObjectServiceContext context;

    public InvokeOperationRequest(
            ObjectServiceContext context,
            String operationName,
            List<Variant> inputArguments)
    {
        this.id = context.getObjectId().getId();
        this.operationName = operationName;
        this.inputArguments = inputArguments;
        this.context = context;
    }

    public String getId() {
        return id;
    }

    public String getOperationName() {
        return operationName;
    }

    public List<Variant> getInputArguments() {
        return inputArguments;
    }

    public ObjectServiceContext getContext() {
        return context;
    }
}

File: opcua-webservice-node/src/server/service/message/digitialtwin/InvokeOperationRequest.ts
import { UaVariant } from "opcua-webapi-ts";
import { ObjectServiceContext } from "../../../types/digitaltwin/ObjectServiceContext";

export class InvokeOperationRequest {
    private readonly _id: string;
    private readonly _operationName: string;
    private readonly _inputArguments: Array<UaVariant>;
    private readonly _context: ObjectServiceContext;

    constructor(
        context: ObjectServiceContext,
        operationName: string,
        inputArguments: Array<UaVariant>) {
        this._id = context.objectId.id;
        this._operationName = operationName;
        this._inputArguments = inputArguments;
        this._context = context;
    }

    get id(): string {
        return this._id;
    }

    get operationName(): string {
        return this._operationName;
    }

    get inputArguments(): Array<UaVariant> {
        return this._inputArguments;
    }

    get context(): ObjectServiceContext {
        return this._context;
    }
}